src/UhdmAst.cpp

```diff
@@ -645,38 +668,34 @@ AstNodeDType* getDType(FileLine* fl, vpiHandle obj_h, UhdmShared& shared) {
         break;
     }
     case vpiArrayVar: {
-        auto array_type = vpi_get(vpiArrayType, obj_h);
-        // todo: static/dynamic/assoc/queue
-        auto rand_type = vpi_get(vpiRandType, obj_h);
-        // todo: rand/randc/notrand
-        AstNodeDType* elementDtypep = nullptr;
-
-        vpiHandle itr = vpi_iterate(vpiReg, obj_h);
-        while (vpiHandle member_h = vpi_scan(itr)) {
-            auto type_h = vpi_handle(vpiTypespec, member_h);
-            if (type_h) {
-                elementDtypep = getDType(fl, type_h, shared);
+        auto typespec_h = vpi_handle(vpiTypespec, obj_h);
+        if (typespec_h) {
+            dtype = getDType(fl, typespec_h, shared);
+        } else {  // Try to get the type from one of the elements
+            auto itr = vpi_iterate(vpiReg, obj_h);
+            auto member_h = vpi_scan(itr);
+            typespec_h = vpi_handle(vpiTypespec, member_h);
+            if (typespec_h) {
+                dtype = getDType(fl, typespec_h, shared);
             } else {
                 auto element_type = vpi_get(vpiType, member_h);
                 if (element_type) {
                     AstBasicDTypeKwd keyword = get_kwd_for_type(element_type);
-                    elementDtypep = new AstBasicDType(fl, keyword);
+                    dtype = new AstBasicDType(fl, keyword);
+                } else {
+                    v3error("Missing typespec for unpacked_array_var");
```

Comment:
Match the VPI/UHDM name here: `array_var`.

Reply:
Fixed